clients/src/main/java/org/apache/kafka/clients/ClientUtils.java

```diff
@@ -70,6 +71,8 @@ public static void closeQuietly(Closeable c, String name, AtomicReference<Throwa
             try {
                 c.close();
             } catch (Throwable t) {
+                if (t instanceof InterruptException)
```

Comment:
I think this might not be quite what we want. When `InterruptException` is thrown, the interrupt flag is explicitly set, so there shouldn't be a need to do it again. We would need to reenable it if we catch `InterruptedException` though. It's not clear what the semantics of this API should be, but I think the main goal is to avoid is losing track of the fact that the interrupt occurred. Here are a couple options that come to mind:

1. If the `InterruptedException` is returned in the atomic reference, then we haven't destroyed evidence of the interrupt, so perhaps we don't need to call `interrupt()`. On the other hand, if the reference is already set, then we must use `interrupt()` or we would lose the interrupt. 
2. Perhaps we need to treat interrupts as a separate concern. For example, perhaps the result of this function could be a boolean indicating whether the close was interrupted. We could come up with an invariant on the interrupt status which applies to swallowing both `InterruptException` and `InterruptedException`. Another option might be to add a third argument to store the possible interrupt exception.

Reply:
Thanks. Think we should not make it too complicated. Suggestion (1) makes sense to me. Will update accordingly.

Reply:
Good point.